radicle-surf: add CHANGELOG.md (#110)
radicle-dev/radicle-git: radicle-surf/CHANGELOG.md

> ## Version 0.9.0

Author: I'm okay with either v0.9 or v1.0. I bumped it only to  0.9.0 as I didn't feel the API is mature enough for 1.0.0, yet.

I will update `Cargo.toml`.

This release consists of a major rewrite of this crate. Its API is overall
simplified and is not compatible with the previous version (v0.8.0). The main
changes include:

- `Browser` is removed. Its methods are implemented directly with `Repository`.

> - `vcs` is removed. No more generic VCS. Only `git` is supported.

Reviewer: ```suggestion
- Git will be the only supported VCS. Any extension points for other VCSes were removed.
```

> - `Ref` and `RefScope` are removed. Re-use `git-ref-format` crate and a new

Reviewer: ```suggestion
- `Ref` and `RefScope` are removed. Re-use the `git-ref-format` crate and a new
```

`glob` for the ref patterns.

> - Added support of `Tree` and `Blob` that correspond to their defintions in git.

Reviewer: ```suggestion
- Added support of `Tree` and `Blob` that correspond to their definitions in Git.
```

- Added two new traits `Revision` and `ToCommit` that make methods flexiable and
still simple to use.

> - The module hierarchy is much more flat than before.

Reviewer: I don't think we need to mention this tbh
```suggestion
```
